plugins/recover.c

```diff
@@ -60,6 +64,85 @@ static struct command_result *after_emergency_recover(struct command *cmd,
 	return command_still_pending(cmd);
 }
 
+static struct command_result *after_restorefrompeer(struct command *cmd,
+					             const char *buf,
+					             const jsmntok_t *params,
+					             void *cb_arg UNUSED)
+{
+	plugin_log(plugin, LOG_DBG, "restorefrompeer called");
+
+	peer_storage_timer = plugin_timer(plugin, time_from_sec(5), do_find_peer_storage, NULL);
+	return command_still_pending(NULL);
+}
+
+static struct command_result *find_peer_storage (void *unused) {
+	peer_storage_timer = NULL;
+
+	struct out_req *req;
+	req = jsonrpc_request_start(plugin, NULL, "restorefrompeer",
+					after_restorefrompeer,
+					&forward_error, NULL);
+
+	return send_outreq(plugin, req);
+}
+
+static void do_find_peer_storage (void *unused) {
+	find_peer_storage(NULL);
+	return;
+}
+
+
+static void do_check_gossip (void *unused) {
+	find_exes_timer = NULL;
+
+	gossmap_refresh(global_gossmap, NULL);
+
+	plugin_log(plugin, LOG_DBG, "Finding our node in gossip");
+
+	struct gossmap_node *n = gossmap_find_node(global_gossmap, &local_id);
+
+	if (n) {
+		for (struct gossmap_chan *c = gossmap_first_chan(global_gossmap);
+	     		c;
+	     		c = gossmap_next_chan(global_gossmap, c)) {
+				struct gossmap_node *x = gossmap_node_byidx(global_gossmap, c->half[1].nodeidx);
+				struct gossmap_node *y = gossmap_node_byidx(global_gossmap, c->half[0].nodeidx);
+				struct node_id a, b;
+				gossmap_node_get_id(global_gossmap, x, &a);
+				gossmap_node_get_id(global_gossmap, y, &b);
+
```

Comment:
This iterates the entire graph!  You only want to do this node, so `gossmap_nth_chan`.  That also tells you which half the node is....